### ISSUE TITLE
Zotero 5 incompatible

### ISSUE THREAD

**Issue by GCamp6** [retorquere/zotero-report-customizer#22]
With the Zotero 5.0-beta.222+a56af4a10 this addon is disabled due to incompatibility. Plan for upgrading it?

Thanks

**Comment by retorquere** [comment]
I'm considering it, but I'm waiting to hear what their post-5.0 plans are. 5.0 is a transient version as there are already plans to port Zotero to Electron. If 5.0 is short-lived, I'll target the Electron version directly.

**Comment by GCamp6** [comment]
Ok, thanks for the clarification! In the meanwhile I suppose we can use the [Online Customizer](http://jasonpriem.org/projects/report_cleaner.php).

Cheers

**Comment by gracile-fr** [comment]
@retorquere : what's difficult here? Does ZRC work with Z4 Standalone (I don't recall).

**Comment by retorquere** [comment]
ZRC works with standalone. It's probably not hard to port, but bbt is a full time effort right now.

**Comment by gracile-fr** [comment]
Yes, obviously you don't have time for this. But I'd like to have a look. I've changed maxVersion of install.rdf to 5.0 and changed overlay.xul to have ZRC menu item in Tools menu. But when I run Zotero.ReportCustomizer.openPreferenceWindow(); I got <code>Zotero.ReportCustomizer is undefined</code> error.

**Comment by retorquere** [comment]
I've done a quick update to the new setup I'll be using for Z5; it's both a cleanup and a prep for the changes that will come again when Z5 ports to Electron. When the BBT port is done I'll split off the build infra to a separate npm package for re-use, but right now it's just a stripped copy from the bbt scripts.

You can build the xpi (after installing node and running `npm i`) using `npm run build`.

Basic debugging is now:

1. Start Zotero with the `-ZoteroDebugText` flag and redirect the output to a file
2. Let it start up, then stop Zotero
3. In the logfile you created in 1., look for `load failed`

**Comment by atanasj** [comment]
Hi. Sorry, but my skills / knowledge in the area you guys are speaking about is limited.... but, is ZRC now useable in Z5?

**Comment by retorquere** [comment]
It is not.

**Comment by atanasj** [comment]
Thanks for your response. ZRC is a super useful tool (at least for my workflow). Is this something you will update for Z5?

**Comment by retorquere** [comment]
I'm not sure. A lot has changed under the hood in Zotero 5 and the changes seem to make it fairly difficult to do this round.

**Comment by atanasj** [comment]
No worries. Well, I certainly hope this can be updated for Z5 and onwards. I imagine that this kind of work takes a significant amount of effort, so not surprised you wouldn't be rushing to update a moving target (with the potential other changes Zotero may have in the future). Even still, thanks for creating this add-in, and I will continue holding out on Z4 as ZRC is just too good to give up!

**Comment by retorquere** [comment]
What do you use it for if I may ask? Wouldn't generating a bibliography do much the same?

**Comment by atanasj** [comment]
I was using it to create an annotated bibliography / systematic review of sorts. The report would just extract the document attributes I wanted, and the annotations from the document via zotfile would be brought into the report. 
I use Scrivener, so custom metadata would hold the relevant bits of info in a note I would make on the pdf. 

However, I bit the bullet, and updated  to Z5, and noticed that zotfile now supports PDF expert. So while I can't get the report the way I want, and the document title no longer hyperlinks back to the zotero record, the extracted annotations now link back to the exact spot in the linked pdf, which is pretty cool.

**Comment by retorquere** [comment]
Annotations via zotfile?

The reason I'm asking is that I may be able to repurpose the bibliography pipeworks to redo the report customizer. The links to the references might be hard to do however.

**Comment by atanasj** [comment]
Yep. Zotfile time extract the annotations, and then add the custom report to Scrivener. I can then use the data for writing, or export to csv to report on. 
If you can get ZRC back up, that'd be super!
